### DISCUSSION TITLE
How to split a branch?

### DISCUSSION THREAD

**Discussion by zyberzebra** [VirtusLab/git-machete#1314] [Q&A] [answered]
## Let's say I have this:
The numbers are commits.

Feature-Branch 1:  1---2
Feature-Branch 2:      3---4
Feature-Branch 3:          5---6

## And I would like:
 
Feature-Branch 1: 1---2
Feature-Branch 2:      3
Feature-Branch 3:       4
Feature-Branch 4:        5---6

Thanks for help!

**Comment by PawelLipski** [comment, marked as answer]
Hey, I think you could just edit `.git/machete` (with `git machete edit` or `git machete add`) and reset the branches to point to the right commits... so something like:

```
git checkout Feature-Branch-3
git machete add Feature-Branch-4   # or manually, with git machete edit

git checkout Feature-Branch-2
git reset --keep commit-3   # --keep is similar but safer than --hard

git checkout Feature-Branch-3
git reset --keep commit-4
```
